File: src/frontend/src/api/schemas/principle.ts
import { z } from 'zod';

export const PrincipleSchema = z.object({
  id: z.string().uuid(),
  nodeId: z.string().uuid(),
  title: z.string().min(1),
  content: z.string().min(1),
  order: z.number().int(),
  isOverride: z.boolean(),
  createdAt: z.string().datetime(),
  updatedAt: z.string().datetime(),
});

export type Principle = z.infer<typeof PrincipleSchema>;

export const EffectivePrincipleSchema = z.object({
  principle: PrincipleSchema,
  sourceNodeId: z.string().uuid(),
  isInherited: z.boolean(),
  isOverridden: z.boolean(),
});

export type EffectivePrinciple = z.infer<typeof EffectivePrincipleSchema>;

File: src/backend/AgenticCompany.Core/Entities/Principle.cs
namespace AgenticCompany.Core.Entities;

public class Principle
{
    public Guid Id { get; set; }
    public Guid NodeId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public int Order { get; set; }
    public bool IsOverride { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Node Node { get; set; } = null!;
}
